test/recipes/70-test_sslmessages.t

```diff
@@ -0,0 +1,352 @@
+              "Server name handshake test (client)");
+
+#Test 9: Server name handshake (server support only)
+$proxy->clear();
+$proxy->clientflags("-no_tls1_3");
+$proxy->serverflags("-servername testhost");
+$proxy->start();
+checkhandshake($proxy, checkhandshake::DEFAULT_HANDSHAKE,
+               checkhandshake::DEFAULT_EXTENSIONS,
+              "Server name handshake test (server)");
+
+#Test 10: Server name handshake (client and server)
+$proxy->clear();
+$proxy->clientflags("-no_tls1_3 -servername testhost");
+$proxy->serverflags("-servername testhost");
+$proxy->start();
+checkhandshake($proxy, checkhandshake::DEFAULT_HANDSHAKE,
+              checkhandshake::DEFAULT_EXTENSIONS
+              | checkhandshake::SERVER_NAME_CLI_EXTENSION
+              | checkhandshake::SERVER_NAME_SRV_EXTENSION,
+              "Server name handshake test");
+
+#Test 11: ALPN handshake (client request only)
+$proxy->clear();
+$proxy->clientflags("-no_tls1_3 -alpn test");
+$proxy->start();
+checkhandshake($proxy, checkhandshake::DEFAULT_HANDSHAKE,
+               checkhandshake::DEFAULT_EXTENSIONS
+               | checkhandshake::ALPN_CLI_EXTENSION,
+              "ALPN handshake test (client)");
+
+#Test 12: ALPN handshake (server support only)
+$proxy->clear();
+$proxy->clientflags("-no_tls1_3");
+$proxy->serverflags("-alpn test");
+$proxy->start();
+checkhandshake($proxy, checkhandshake::DEFAULT_HANDSHAKE,
+               checkhandshake::DEFAULT_EXTENSIONS,
+              "ALPN handshake test (server)");
+
+#Test 13: ALPN handshake (client and server)
+$proxy->clear();
+$proxy->clientflags("-no_tls1_3 -alpn test");
+$proxy->serverflags("-alpn test");
+$proxy->start();
+checkhandshake($proxy, checkhandshake::DEFAULT_HANDSHAKE,
+              checkhandshake::DEFAULT_EXTENSIONS
+              | checkhandshake::ALPN_CLI_EXTENSION
+              | checkhandshake::ALPN_SRV_EXTENSION,
+              "ALPN handshake test");
+
+#Test 14: SCT handshake (client request only)
+$proxy->clear();
+#Note: -ct also sends status_request
+$proxy->clientflags("-no_tls1_3 -ct");
+$proxy->serverflags("-status_file "
+                    .srctop_file("test", "recipes", "ocsp-response.der"));
+$proxy->start();
+checkhandshake($proxy, checkhandshake::OCSP_HANDSHAKE,
+              checkhandshake::DEFAULT_EXTENSIONS
+              | checkhandshake::SCT_CLI_EXTENSION
+              | checkhandshake::STATUS_REQUEST_CLI_EXTENSION
+              | checkhandshake::STATUS_REQUEST_SRV_EXTENSION,
+              "SCT handshake test (client)");
+
+#Test 15: SCT handshake (server support only)
+$proxy->clear();
+#Note: -ct also sends status_request
+$proxy->clientflags("-no_tls1_3");
+$proxy->serverflags("-status_file "
+                    .srctop_file("test", "recipes", "ocsp-response.der"));
+$proxy->start();
+checkhandshake($proxy, checkhandshake::DEFAULT_HANDSHAKE,
+              checkhandshake::DEFAULT_EXTENSIONS,
+              "SCT handshake test (server)");
```

Comment:
More indentation that's off by one.  I'll stop now, just remarking that it's the same in the rest of the file.